pootle/apps/pootle_translationproject/models.py

```diff
@@ -277,6 +277,9 @@ def save(self, *args, **kwargs):
         else:
             self.abs_real_path = None
         super(TranslationProject, self).save(*args, **kwargs)
+        if self.directory.tp_id != self.pk:
+            self.directory.tp = self
+            self.directory.save()
```

Comment:
k - there is the remotest chance of a tp.directory which doesnt have the tp set

if you want to add a migration for this please do